job/ipxe.go

```diff
@@ -79,7 +79,12 @@ func (j Job) serveBootScript(ctx context.Context, w http.ResponseWriter, name st
 
 	s.Echo("Packet.net Baremetal - iPXE boot")
 
-	fn(j, s)
+	// the trace id is enough to find otel traces in most systems
+	if sc := span.SpanContext(); sc.IsSampled() {
+		s.Echo("Debug Trace ID: " + sc.TraceID().String())
```

Comment:
We add traceparent to the kernel commandline in the ipxe scripts as well, so this is about printing something you can copy/paste into your tracing GUI to find the trace. I'm working on OSIE now to add code to pick up cmdline and do as you suggest :)

Reply:
I meant for ipxe to use the variable in a header for the fetches of the kernel and initrd files from nginx, presumably then the instrumented nginx/ingress would emit those traces.

Reply:
Oh!! I didn't even think about that. I'll look up how to set headers in ipxe and see about adding that :) That would be so cool

Reply:
looks like ipxe doesn't offer a way to set headers so maybe I'll teach boots http to look for a query param

Reply:
boots doesn't serve the kernel/initird so thats wasted effort. The initrd/kernel files are behind regular ol http servers. We use nginx in sandbox and in EM.

Reply:
> looks like ipxe doesn't offer a way to set headers so maybe I'll teach boots http to look for a query param

Are you saying this from reading the ipxe docs? If so, I wouldn't trust it. The docs are accurate for things documented but iirc somethings just aren't documented, for example http auth which looking through the ipxe code base shows its supported yet there's no info in the docs page.

Reply:
yep. I did see the basic auth support and thought about  abusing that. I still like the query param path. that will make the traceparent show up in the nginx logs and we can later come around and find a way to pump them into the telemetry system

but it can wait :) I doubt this is a common mode of failure?

Reply:
if params works then sure lets start with that then. Actually we have had failures fetching these files from time to time and have logging alerts setup to catch them.